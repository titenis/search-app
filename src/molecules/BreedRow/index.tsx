import TableCell from '@material-ui/core/TableCell';
import { Highlightable } from 'atoms/Highlightable';
import LinkIcon from '@material-ui/icons/Link';
import TableRow from '@material-ui/core/TableRow';
import React, { FC } from 'react';
import { ICatBreed } from 'api/types';

export const BreedRow: FC<ICatBreed> = ({
  name,
  origin,
  life_span,
  temperament,
  weight,
  wikipedia_url,
}) => (
  <TableRow>
    <TableCell component="th" scope="row">
      <Highlightable>{name}</Highlightable>
    </TableCell>
    <TableCell align="right">
      <Highlightable>{origin}</Highlightable>
    </TableCell>
    <TableCell align="right">{life_span}</TableCell>
    <TableCell align="right">
      <Highlightable>{temperament}</Highlightable>
    </TableCell>
    <TableCell align="right">{weight.metric}</TableCell>
    <TableCell align="right">
      <a target="_blank" href={wikipedia_url} rel="noreferrer">
        <LinkIcon />
      </a>
    </TableCell>
  </TableRow>
);
