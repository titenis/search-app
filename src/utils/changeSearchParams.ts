export const changeSearchParams = (
  params: URLSearchParams,
  key: string,
  value?: string,
): URLSearchParams => {
  if (value) {
    params.set(key, value);
  } else {
    params.delete(key);
  }

  return params;
};
