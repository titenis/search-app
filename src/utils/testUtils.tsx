import React, { FC, ReactElement } from 'react';

import { render, RenderResult } from '@testing-library/react';
import { createMemoryHistory, History } from 'history';
import { Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { theme } from 'theme';
import { QueryClient, QueryClientProvider } from 'react-query';
import { renderHook, RenderHookResult } from '@testing-library/react-hooks';

export const Providers = (history: History): FC => ({ children }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        cacheTime: 0,
      },
    },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <Router history={history}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </Router>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  { history = createMemoryHistory(), ...restOptions } = {},
): RenderResult =>
  render(ui, {
    wrapper: Providers(history),
    ...restOptions,
  });

export const customRenderHook = <P, R>(
  hook: (props?: P) => R,
  { history = createMemoryHistory(), ...options } = {},
): RenderHookResult<P, R> =>
  renderHook(hook, {
    wrapper: Providers(history),
    ...options,
  });

export * from '@testing-library/react';

export { customRender as render };
export { customRenderHook as renderHook };
