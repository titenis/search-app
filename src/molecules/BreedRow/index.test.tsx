import React from 'react';
import { BreedRow } from 'molecules/BreedRow/index';
import { Abyssian } from 'mock/data';
import { render } from 'utils/testUtils';

describe('BreedRow', () => {
  it('should render', () => {
    const { container } = render(
      <table>
        <tbody>
          <BreedRow {...Abyssian} />
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();
  });
});
