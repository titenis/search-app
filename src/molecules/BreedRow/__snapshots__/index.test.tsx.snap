// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedRow should render 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root"
          role="cell"
          scope="row"
        >
          <span
            data-testid="highlighter"
          >
            <span
              class=""
            >
              Abyssinian
            </span>
          </span>
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          <span
            data-testid="highlighter"
          >
            <span
              class=""
            >
              Egypt
            </span>
          </span>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          14 - 15
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          <span
            data-testid="highlighter"
          >
            <span
              class=""
            >
              Active, Energetic, Independent, Intelligent, Gentle
            </span>
          </span>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          3 - 5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          <a
            href="https://en.wikipedia.org/wiki/Abyssinian_(cat)"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              />
            </svg>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
